mlrun/model_monitoring/db/stores/sqldb/sql_store.py

```diff
@@ -462,31 +461,31 @@ def update_last_analyzed(
         self._update(
             attributes={mm_schemas.SchedulingKeys.LAST_ANALYZED: last_analyzed},
             table=self.MonitoringSchedulesTable,
-            **application_filter_dict,
+            criteria=criteria,
         )
 
-    def _delete_last_analyzed(self, endpoint_id: str = "", application_name: str = ""):
+    def _delete_last_analyzed(
+        self, endpoint_id: str, application_name: typing.Optional[str] = None
+    ) -> None:
         self._init_monitoring_schedules_table()
-
-        application_filter_dict = self.filter_endpoint_and_application_name(
-            endpoint_id=endpoint_id, application_name=application_name
-        )
-
+        criteria: list[BinaryExpression] = [
+            self.MonitoringSchedulesTable.endpoint_id == endpoint_id
+        ]
+        if application_name is not None:
+            criteria.append(
+                self.MonitoringSchedulesTable.application_name == application_name
+            )
         # Delete the model endpoint record using sqlalchemy ORM
-        self._delete(table=self.MonitoringSchedulesTable, **application_filter_dict)
+        self._delete(table=self.MonitoringSchedulesTable, criteria=criteria)
 
-    def _delete_application_result(
-        self, endpoint_id: str = "", application_name: str = ""
-    ):
+    def _delete_application_result(self, endpoint_id: str) -> None:
```

Comment:
please revert the option to filter on `application_name` as well, as we spoke.

Reply:
Done in 5a8a160